test/dynamics/solvers/test_lanczos.py

```diff
@@ -0,0 +1,90 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""
+Direct tests of helper functions in lanczos.py
+"""
+
+import numpy as np
+from scipy.linalg import expm
+from qiskit_dynamics.solvers.lanczos import (
+    lanczos_basis,
+    lanczos_eigh,
+    lanczos_expm,
+    jax_lanczos_basis,
+    jax_lanczos_eigh,
+    jax_lanczos_expm,
+)
+from ..common import QiskitDynamicsTestCase, TestJaxBase
+
+
+class TestLanczos(QiskitDynamicsTestCase):
+    """Tests for lanczos.py"""
+
+    def lanczos_functions(self):
```

Comment:
This method doesn't need to be here - it can just be part of the `setUp` function. For the JAX version of the class, you can make the setup like:
```
def setUp(self):
     super().setUp()
     self.basis = jax_lanczos_basis
     self.eigh = jax_lanczos_eigh
     self.expm = jax_lanczos_expm
```

I've done this in other places in Dynamics.